command/metadata.go

```diff
@@ -14,12 +15,16 @@ import (
 func metadataGenerators(conf *config.Config) []*metadata.Generator {
 	generators := make([]*metadata.Generator, 0, len(conf.MetadataPlugins))
 
-	workdir := pluginutil.PluginWorkDir()
+	workdir := filepath.Join(pluginutil.PluginWorkDir(), "mackerel-metadata")
 	for name, pluginConfig := range conf.MetadataPlugins {
+		workdir := workdir
+		if dir := lookupPluginWorkDir(pluginConfig.Command.Env); dir != "" {
+			workdir = dir
+		}
 		generator := &metadata.Generator{
 			Name:      name,
 			Config:    pluginConfig,
-			Cachefile: filepath.Join(workdir, "mackerel-metadata", name),
+			Cachefile: filepath.Join(workdir, name),
```

Comment:
Is it ok to not append `mackerel-metadata/` to the path when `MACKEREL_PLUGIN_WORKDIR` is configured per plugin?
`filepath.Join(workdir, name)` may conflict with the plugin executable?

Reply:
The temporary file of metric plugin is stored at `MACKEREL_PLUGIN_WORKDIR/pluginname` (https://github.com/mackerelio/go-mackerel-plugin/blob/7d51210247a3f1a73fdd43098cda470b328cb58f/mackerel-plugin.go#L177) so I think this is consistent. The value is not expected to be set to the bin directory.

Reply:
I understand!

Reply:
I reconsidered this and prepended `mackerel-plugin-metadata-` to the plugin name. The file without the env variable config is not changed.